packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py

```diff
@@ -0,0 +1,272 @@
+import logging
+from datetime import datetime
+from io import StringIO
+from typing import TYPE_CHECKING, Optional, cast
+
+import httpx
+
+from phoenix.datetime_utils import normalize_datetime
+
+if TYPE_CHECKING:
+    import pandas as pd
+
+from phoenix.client.types.spans import (
+    SpanQuery,
+    SpanQueryRequestBody,
+)
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT_IN_SECONDS = 5
+
+
+class Spans:
+    """
+    Provides methods for interacting with span resources.
+
+    Example:
+        Basic usage:
+        Basic usage:
+            >>> from phoenix.client import Client
+            >>> from phoenix.client.types.spans import SpanQuery
+            >>> client = Client()
+            >>> query = SpanQuery().select("name", "span_id").where("name == 'my-span'")
+            >>> df = client.spans.get_spans_dataframe(query=query)
+
+    """
+
+    def __init__(self, client: httpx.Client) -> None:
+        self._client = client
+
+    def get_spans_dataframe(
+        self,
+        query: Optional[SpanQuery] = None,
+        start_time: Optional[datetime] = None,
+        end_time: Optional[datetime] = None,
+        limit: int = 1000,
+        root_spans_only: Optional[bool] = None,
+        project_name: Optional[str] = None,
+        timeout: Optional[int] = DEFAULT_TIMEOUT_IN_SECONDS,
```

Comment:
I keep forgetting - shouldn't we make these all kwargs?

Reply:
these are all kwargs right now